### PR TITLE
Removed internal functions from API docs and module.

### PR DESCRIPTION
**Pull request by toolforger**

### CODE REVIEW

**Comment by asmeurer** [comment]
trigintegrate and deltaintegrate should also be removed from `__init__.py`

**Comment by toolforger** [comment]
Will do once the current Travis run completes (I need to see whether the expected Sphinx failure will happen or not, I can't test Sphinx stuff locally).

**Comment by toolforger** [comment]
trigintegrate and deltaintegrate are out of __init__.py and out of API Reference.

**Comment by asmeurer** [comment]
I think they can stay in the docs somewhere. Maybe we should more clearly separate user functions and internal functions in the docs.

**Comment by toolforger** [comment]
For internal stuff, my thinking is that the docstrings should not be put into the HTML.
Because:
1) It has no advantages. Python coders will know what a docstring is and where to find it.
2) It has the disadvantage that the HTML will be 90% docstring copies, making it easy to overlook those 10% that are not in a docstring.
3) Those 90% docstring copies will even make people unsure, because they can't be confident that the HTML is indeed a verbatim copy of the docstrings (until they discover the doc generation mechanism, but most people don't care about it, nor should they have to).

I have to admit that I took a quick look at the HTML, found "aw it's just docstrings", and never took a second look. I bet I'm missing tons of stuff, but I don't want to spend the time *finding* the interesting bits.

**Comment by asmeurer** [comment]
I've found it useful to have the internal functions in the html docs before. But they should be separate, as it's otherwise confusing to users.

**Comment by toolforger** [comment]
Test failure is due to a Theano test that stalled.

**Comment by toolforger** [comment]
@asmeurer once a PR or issue comes up about documenting internal functions, ping me and I'll have my suggestions so that both your and my doc-reading style will work well.

**Comment by skirpichev** [comment]
> But they should be separate, as it's otherwise confusing to users.

For sure, references to missing documentation (e.g. to manualintegrate, or to ratint) can't confuse users...  @toolforger, is there any chance that you will read the code first?

**Comment by asmeurer** [comment]
Sergey, your comments are welcome, but please stop being antagonistic towards people.

**Comment by skirpichev** [comment]
> your comments are welcome
> but please stop

Looks like a contradiction, isn't?

> being antagonistic towards people

Why I should do care anymore about your double standards?

**Comment by asmeurer** [comment]
Let me put it in another way. You are welcome to comment on pull request, but be nice to people. If you can't figure out how to do both, then do neither.  

> Why I should do care anymore about your double standards?

If you see someone else behaving inappropriately, please bring it to my attention.

**Comment by moorepants** [comment]
@skirpichev I'm glad to see that you may be interested in participating SymPy again, but why come back and be antagonistic towards @toolforger from the get go? The whole reason you stopped participating is because we asked you not to do this. For example, how is this statement useful to anyone:

> is there any chance that you will read the code first?

My initial perception is that you are trying to antagonize him. No one here is not "reading the code". Why not just omit statements like that in your responses? Or you can rephrase it to be polite, maybe:

> Joachim, I agree with Aaron that we should leave the docs in. I think the code in <some file> on lines x through y may shed some light on this. When I read those lines, I interpret them as <blank> and I would make the decision to do <blank> based on the following reasoning. Let me know what you think.

When you write something like what you wrote I think many people may perceive it like so:

> That's an odd statement and seems antagonistic. I've read the code many times. I'm not sure what code he's exactly referring to or why he thinks I'm not reading the code. I know I don't understand all of the code completely but I'm trying. The statement makes me feel like I'm not trying or that my work does not meet his standards. I'd much prefer help understanding the code instead of being belittled for not understanding it the same way he does.

**Comment by certik** [comment]
I agree 100% with @asmeurer and @moorepants. Thanks Jason for taking the time to provide examples of how the same thing can be said politely as well as how the original comment by Sergey is interpreted.

@skirpichev please read @moorepants carefully and think about it hard. If you continue attacking people, you will leave us no other option than to block you at github.

**Comment by toolforger** [comment]
Confirming @moorepants: that last paragraph is a 100% accurate description of how I felt when I read his comment.

**Comment by skirpichev** [comment]
> interested in participating SymPy again

No.

> The whole reason you stopped participating

Doesn't make sense any real participation in dying community.  That's the reason, in short.

> My initial perception is that you are trying to antagonize him.

No.

> No one here is not "reading the code".

Wrong.  [Here](https://github.com/sympy/sympy/pull/9114#issuecomment-78131746) is an example who and good reasons why I think so.

It's easy to admit.  It's easy to improve for contributor.

**Comment by skirpichev** [comment]
> block you at github

Sure, you should.

I can't act anymore in a primadonna-like style "don't block me, or I'll leave".

**Comment by asmeurer** [comment]
>> interested in participating SymPy again

> No

Then what are you doing?

**Comment by skirpichev** [comment]
> Then what are you doing?

Watching how you waste your time in teaching people how to do polite comments, instead of doing some constructive job for sympy ;)  E.g. review PRs.

**Comment by skirpichev** [comment]
BTW

> If you see someone else behaving inappropriately, please bring it to my attention. 

Oh :(
https://github.com/sympy/sympy/pull/9088#issuecomment-77912044

**Comment by certik** [comment]
> Doesn't make sense any real participation in dying community. That's the reason, in short.

> Watching how you waste your time in teaching people how to do polite comments, instead of doing some constructive job for sympy ;) E.g. review PRs.

Sergey, it's you who is wasting our (and your!) time. In SymPy we help anybody who is genuinely interested in contributing. If you think you can do things better, then go ahead and show us how to run things better, we'll be happy to learn.

**Comment by moorepants** [comment]
@toolforger has definitely been guilty of not being polite and fueling the antagonistic conversation between you two, including the comment you cite. He has been told the same thing you have. The difference is that he seems to recognize his mistakes and is trying to change his behavior whereas you don't.

**Comment by skirpichev** [comment]
> Sergey, it's you who is wasting our (and your!) time.

Ondrej, how?!

All I did - simple comment that points to flaws of the current patch.  Simple and clear solution (suggested in the old PR too) - revert all changes in docs.

> If you think you can do things better, then go ahead and show us how to run things better

How I can?!  You are just about blocking me...

But if sympy community is about to be friendly and welcoming - why just not take constructive part of some reply and ignore everything that seems to be non-polite for you?

**Comment by skirpichev** [comment]
> The difference is that he seems to recognize his mistakes and is trying to change his behavior whereas you don't.

"And no matter what you think about it" ;)

**Comment by toolforger** [comment]
Am 10.03.2015 um 22:38 schrieb Sergey B Kirpichev:
>> Sergey, it's you who is wasting our (and your!) time.
>
> Ondrej, how?!
>
> All I did - simple comment that points to flaws of the current
> patch.

That part was okay.

> Simple and clear solution (suggested in the old PR too) - revert all
 > changes in docs.

Simple - yes.
Clear - yes.
Correct - no.

>> If you think you can do things better, then go ahead and show us how to run things better
>
> How I can?!  You are just about blocking me...

You're being dishonest.
You started a fork (omg), so you know fully well how to show us how to 
run things better.

> But if sympy community is about to be friendly and welcoming

Well, if you can't be that, you're obviously unfit for being in that 
community.

> - why just not take constructive part of some reply and ignore
 > everything that seems to be non-polite for you?

I did take the constructive part.

**Comment by skirpichev** [comment]
On Tue, Mar 10, 2015 at 03:41:47PM -0700, Joachim Durchholz wrote:
>    > Simple and clear solution (suggested in the old PR too) - revert all
>    > changes in docs.
> 
>    Simple - yes.
>    Clear - yes.
>    Correct - no.

For sure, if you don't read some code (as in this case) -
you should quickly revert all changes you made.

> You're being dishonest.

Wow.  Some time ago, you did tens unconstructive comments
before I locked you from conversation in one PR.

I did a single comment and Ondrej already talking about blocking
me on github...  Is this ... honest?

>    You started a fork (omg), so you know fully well how to show us how to
>    run things better.

Sure, you know better why I started this fork...

>    > But if sympy community is about to be friendly and welcoming
> 
>    Well, if you can't be that, you're obviously unfit for being in that
>    community.

I'm pointing on the problem (you gays can consider this to be
a feature, of course), that claims about "friendly and
welcoming" - is a lie.

**Comment by certik** [comment]
> (you gays can consider this to be a feature, of course)

I really hope you meant to write "guys" instead of "gays".

@skirpichev I will not block anyone for unconstructive comments. But you are attacking people (besides being unconstructive) and that is not acceptable.

> I did a single comment and Ondrej already talking about blocking me on github... Is this ... honest?

You have been warned many times, and you specifically wrote me you do not want to improve. So the next step if you continue this behavior is that I will block you without further warning. The decision will be made by me and it is subjective.

You are right on the line, so if you do no want to be blocked, then do not attack people. If you want more help from us with your communication, then you must be willing to work on this and try to improve, instead of ignoring our help and warnings.

**Comment by skirpichev** [comment]
> I really hope 

Sorry for yet another typo...

> But you are attacking people

No.  You think I'm "attacking people".  It's not the same.

The problem is you consider yourself to be one source of truth.  You can't be wrong - that's part of the problem, despite it seems not all people in the sympy community share same opinion about me (see e.g. [this](https://groups.google.com/forum/?fromgroups=#!topic/sympy/YlZ42TMWFGI) thread).

> besides being unconstructive

Where and why?  Who started this whole thread about politeness and so on?  Am I?

> The decision will be made by me and it is subjective.

BTW, why you can do this decision yourself?  Should you ask first other community members, e.g. @smichr, @mrocklin, @cbm755?  Are @asmeurer only a nominal leader?

> if you do no want to be blocked

I don't care anymore, really.  But I think you shouldn't block people if they say unpleasant things about your project or community members.

Also, if you will do this - please remove me from aboutus.rst and AUTHORS as well.

**Comment by certik** [comment]
You can read the structure of SymPy here:

https://groups.google.com/d/topic/sympy/i-XD15syvqs/discussion

Aaron is the leader about any code decisions, and I never override him. In things like this, Aaron has a say first, but ultimately the final decision is by me. But Aaron and I are in agreement about this so far, so there is no issue there either.

> But I think you shouldn't block people if they say unpleasant things about your project or community members.

I agree on this, and I am not. You are welcome to criticize me as much as you want.

> Also, if you will do this - please remove me from aboutus.rst and AUTHORS as well.

If that is what you want, you can always send me an email anytime and I can do that.

**Comment by skirpichev** [comment]
> But Aaron and I are in agreement about this so far, so there is no issue there either.

Blaming about "non-politeness" is easy, esp. neither you nor Aaron don't want to listen to counterarguments (sorry, by that was my impression from private conversations).  Writing good code or doing constructive reviews - is not.

> You are welcome to criticize me as much as you want.

And then we ban you...

> If that is what you want

Sure.  Logically - this should be your next "friendly & welcoming" step.  Maybe next time people will think about investing time in your pet project.

**Comment by certik** [comment]
@skirpichev I am sorry if you feel your time was lost. I truly am, I know how it feels to invest my time and energy into projects that turn out not the way I liked. However, attacking everybody around you (including myself now) is not the solution. Please stop.

**Comment by skirpichev** [comment]
> attacking everybody around you

Oh, and that is all your "You are welcome to criticize me"...

**Comment by toolforger** [comment]
attack != criticize

**Comment by skirpichev** [comment]
Oh, dear.  Just call any critique "personal attacking" - and ban people.

**Comment by toolforger** [comment]
Nah, not any critique. Just that which is a personal attack.

**Comment by skirpichev** [comment]
> Just that which is a personal attack.

Sure, with this useful definition you can ban that you don't like.

**Comment by certik** [comment]
Critique is your first part: that we should review more and talk less.
Attack is when you say things about pet projects and hoping people will
think twice before participating (it's definitely not critique). You can
critique as much as you want.

Let's get back to work on this thread.

**Comment by skirpichev** [comment]
>    Let's get back to work on this thread.

You have clear answers from @toolforger that "I did take the
constructive part."  Despite the fact that the documentation was
removed and (new) broken links are still in place...

(BTW, there are lot of them in other places as well.  Definitely, you
 should try to run sphinx-build in nit-picky mode.)

**Comment by toolforger** [comment]
It would be nice if you could open an issue for the broken links.

**Comment by asmeurer** [comment]
He's probably referring to things like http://docs.sympy.org/latest/modules/integrals/integrals.html#sympy.integrals.trigintegrate (i.e., links for docs that were removed are now broken, or at least will be when the docs are rebuilt).

**Comment by skirpichev** [comment]
> He's probably referring to things like

[I'm about](https://github.com/sympy/sympy/pull/9114#issuecomment-78131746) manualintegrate and ratint (use simple search [here](http://docs.sympy.org/latest/modules/integrals/integrals.html)), but your example "works" too.

**Comment by asmeurer** [comment]
@toolforger can you make a pull request that reverts the changes in this pull request to doc/src/modules/integrals/integrals.rst

**Comment by toolforger** [comment]
I can - is this really the best/easiest way forward? (I'm not in a position to judge, just curious.)

**Comment by asmeurer** [comment]
It's definitely the easiest.

As to the best, those methods being documented there is the way it is done in the last release, so it's not a regression. They were removed without proper discussion.  I'm -1 to their outright removal (and obviously so is Sergey).

**Comment by toolforger** [comment]
Well, with Sergey's opinion I can never tell whether it is in line with SymPy's overall goals or just a reflection of his personal ones. Also, I often find him very hard to understand, and find myself unable to get clarification from him.
I.e. I have no way to tell whether I should care about a particular -1 vote from him or not.

On reconsideration, I see I can't simply put the lines back in. To suppress an import cycle, I had to remove the symbols from the submodule that the ```.rst``` would pick them up from, so I'd have to change the absolute path.
I think I'll have to add new submodule sections, probably fix links as well, and do the testing via Travis because I can't test Sphinx stuff locally. So for the part I'm seeing now, re-adding the symbols is definitely not the easiest way.
I'm definitely not in the best position to fix this: no Sphinx knowledge, no way to do the Sphinx testing locally because Sphinx not installed in my virtualenvs and my system Python doesn't have mpmath. Could somebody else pick this up, please?

**Comment by moorepants** [comment]
Can't you just `pip install sphinx`?

**Comment by toolforger** [comment]
Didn't try... those virtualenvs are just throwaway installs to test SymPy, I'd have to install sphinx in every one of them (and remember to install it again whenever I set up a new one).
I can give it a try. Still, I'd probably be the slowest around to find the links that would need to be updated.

**Comment by skirpichev** [comment]
> Well, with Sergey's opinion I can never tell whether it is in line with SymPy's overall goals or just a reflection of his personal ones.

"anyone with Python on his/her computer can review" (c) your own doc, guys.

Is that just another lie?

> I.e. I have no way to tell whether I should care about a particular -1 vote from him or not.

Sure, I'm not among several bless people in the sympy community all should do care about...

But the whole matter is a simple and trivial regression.  Common sense?  No?

> Also, I often find him very hard to understand, and find myself unable to get clarification from him.

It happens to be only your problem.  At least from answers of other people I can't see that.  Anyone here shares same opinion as @toolforger?

BTW what is hard in the sentence "if you don't read some code (as in this case) - you should quickly revert all changes you made."?  Simply revert your changes in docs (or follow [this](https://github.com/sympy/sympy/pull/9088#issuecomment-77908055) recommendation; i.e. change authomethod -> autofunction).

> To suppress an import cycle, I had to remove the symbols from the submodule that the .rst would pick them up from

You can use suggested change authomethod -> autofunction (I'm not sure if that really needed).  But you should revert removal of the docs.

> I think I'll have to add new submodule sections, probably fix links as well, and do the testing via Travis because I can't test Sphinx stuff locally.

@asmeurer, @certik, @moorepants, do you see that?  Was I incorrect in [this diagnosis](https://github.com/sympy/sympy/pull/9114#issuecomment-78131746)?

No.  He actually didn't read the code, he actually don't build the docs.  Instead of fixing that issue - you ban people!

> I'm definitely not in the best position to fix this: no Sphinx knowledge, no way to do the Sphinx testing locally 

If you don't know something - ask people first, not do claims and not make sentences like "It was broken and needed fixing, no matter what you say".

FYI: documentation building is a standard part of development.  You should be able to follow instructions in the docs (doc/README.txt and so on) and https://github.com/sympy/sympy/wiki/Development-workflow.  If it seems unclear to you how to set up your local machine for development - please ask (btw, maybe development documentation should be improved).  Don't silently ignore the building of docs!  You can do sphinx testing locally, the problem is - you do not.

Unfortunately, this seems to be a pattern.  Recently I commented on some PR (merged by @smichr, AFAIR), where added documentation looks badly (i.e. nobody actually tried to build one and look on it).

Guys, do you see a real problem now or you still prefer to talk about non-politeness and how to ban people?

> Still, I'd probably be the slowest around to find the links that would need to be updated.

If you just revert the removal, as it was repeatedly suggested by several people - no links will be broken, don't care about.

**Comment by toolforger** [comment]
Am 12.03.2015 um 11:32 schrieb Sergey B Kirpichev:

["political" aspects snipped - I disagree but do not think that arguing 
can resolve anything]

>> To suppress an import cycle, I had to remove the symbols from the
>> submodule that the .rst would pick them up from
>
> You can use suggested change authomethod -> autofunction

Agreed, that's the uncontroversial change then.

 > (I'm not
> sure if that really needed).

It will just generate the wrong kind of doc (member function instead of 
package function).
Nothing that a unit test will find, but still wrong.

 > But you should revert removal of the
> docs.

Heh. You can easily try that out yourself.
Just pull the current master and reinsert one of the autofunction lines.
Then, build the docs (should be easy as you say), or push to a 
Travis-enabled github repo and look at the TEST_SPHINX="true" line.

Last time I pushed with such a function, I had that test run error out 
due to Sphinx errors.
I have no reason to believe that this has changed.

>> Still, I'd probably be the slowest around to find the links that
>> would need to be updated.
>
> If you just revert the removal, as it was repeatedly suggested by
> several people - no links will be broken, don't care about.

It won't break links, just the Travis test.
I do not think that's an improvement.

**Comment by skirpichev** [comment]
On Thu, Mar 12, 2015 at 05:20:09AM -0700, Joachim Durchholz wrote:
>    > You can use suggested change authomethod -> autofunction
> 
>    Agreed, that's the uncontroversial change then.

It can't hurt, that's all.

>    It won't break links

This pr break links.  How hard to understand that?

**Comment by skirpichev** [comment]
Trivial regression problem, long and worthless talks about politeness...  And no practical action even after a week.

@certik, @asmeurer is it a SymPy way now?
